internal/resource/config/incomingproxysettings/incoming_proxy_settings_resource.go

```diff
@@ -0,0 +1,279 @@
+				Computed:    true,
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("FIRST", "LAST"),
+					stringvalidator.AlsoRequires(path.MatchRoot("forwarded_host_header_name")),
+				},
+			},
+			"client_cert_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"client_cert_chain_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the chain header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"proxy_terminates_https_conns": schema.BoolAttribute{
+				Description: "Allows you to globally specify that connections to the reverse proxy are made over HTTPS even when HTTP is used between the reverse proxy and PingFederate.",
+				Computed:    true,
+				Optional:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalIncomingProxySettingsFields(ctx context.Context, addRequest *client.IncomingProxySettings, plan incomingProxySettingsResourceModel) {
+
+	addRequest.ForwardedIpAddressHeaderName = plan.ForwardedIpAddressHeaderName.ValueStringPointer()
+
+	if internaltypes.IsDefined(plan.ForwardedIpAddressHeaderIndex) {
+		addRequest.ForwardedIpAddressHeaderIndex = plan.ForwardedIpAddressHeaderIndex.ValueStringPointer()
+	}
+	addRequest.ForwardedHostHeaderName = plan.ForwardedHostHeaderName.ValueStringPointer()
+
+	if internaltypes.IsDefined(plan.ForwardedHostHeaderIndex) {
+		addRequest.ForwardedHostHeaderIndex = plan.ForwardedHostHeaderIndex.ValueStringPointer()
+	}
+	addRequest.ClientCertSSLHeaderName = plan.ClientCertSSLHeaderName.ValueStringPointer()
+
+	addRequest.ClientCertChainSSLHeaderName = plan.ClientCertChainSSLHeaderName.ValueStringPointer()
+
+	addRequest.ProxyTerminatesHttpsConns = plan.ProxyTerminatesHttpsConns.ValueBoolPointer()
+
+}
+
+// Metadata returns the resource type name.
+func (r *incomingProxySettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_incoming_proxy_settings"
+}
+
+func (r *incomingProxySettingsResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *incomingProxySettingsResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	var plan *incomingProxySettingsResourceModel
+	req.Plan.Get(ctx, &plan)
```

Comment:
You can record the diagnostics of the Plan.Get call, with something like `resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)`